crits/backdoors/handlers.py

```diff
@@ -0,0 +1,423 @@
+        }
+
+        #objects
+        objects = backdoor.sort_objects()
+
+        #relationships
+        relationships = backdoor.sort_relationships("%s" % analyst, meta=True)
+
+        # relationship
+        relationship = {
+            'type': 'Backdoor',
+            'value': backdoor.id
+        }
+
+        #comments
+        comments = {'comments': backdoor.get_comments(),
+                    'url_key': backdoor.id}
+
+        #screenshots
+        screenshots = backdoor.get_screenshots(analyst)
+
+        # favorites
+        favorite = is_user_favorite("%s" % analyst, 'Backdoor', backdoor.id)
+
+        # services
+        service_list = get_supported_services('Backdoor')
+
+        # analysis results
+        service_results = backdoor.get_analysis_results()
+
+        args = {'objects': objects,
+                'relationships': relationships,
+                'relationship': relationship,
+                'subscription': subscription,
+                'favorite': favorite,
+                'service_list': service_list,
+                'service_results': service_results,
+                'screenshots': screenshots,
+                'backdoor': backdoor,
+                'backdoor_id': id_,
+                'comments': comments}
+    return template, args
+
+def add_new_backdoor(name, version=None, aliases=None, description=None,
+                     source=None, source_method=None, source_reference=None,
+                     campaign=None, confidence=None, analyst=None,
+                     bucket_list=None, ticket=None):
+    """
+    Add an Backdoor to CRITs.
+
+    :param name: The name of the backdoor.
+    :type name: str
+    :param version: Version of the backdoor.
+    :type version: str
+    :param aliases: Aliases for the backdoor.
+    :type aliases: list or str
+    :param description: Description of the backdoor.
+    :type description: str
+    :param source: Name of the source which provided this information.
+    :type source: str
+    :param source_method: Method of acquiring this data.
+    :type source_method: str
+    :param source_reference: A reference to this data.
+    :type source_reference: str
+    :param campaign: A campaign to attribute to this backdoor.
+    :type campaign: str
+    :param confidence: Confidence level in the campaign attribution.
+    :type confidence: str ("low", "medium", "high")
+    :param analyst: The user adding this backdoor.
+    :type analyst: str
+    :param bucket_list: Buckets to assign to this backdoor.
+    :type bucket_list: str
+    :param ticket: Ticket to assign to this backdoor.
+    :type ticket: str
+    :returns: dict with keys:
+              "success" (boolean),
+              "message" (str),
+              "id" (str),
+              "object" (if successful) :class:`crits.backdoors.backdoor.Backdoor`
+    """
+
+    retVal = {'success': False, 'message': ''}
+
+    if isinstance(source, basestring):
```

Comment:
Might add an elif here in the event source is an EmbeddedSource object in which case you need to make a list as well. Currently only handles string and list for source.